x-pack/plugins/apm/server/routes/service_map/transform_service_map_responses.ts

```diff
@@ -98,9 +135,17 @@ export function transformServiceMapResponses({
     if (!node.id || map[node.id]) {
       return map;
     }
-
+    const outboundConnectionExists = allConnections.some(
+      (con) =>
+        SPAN_DESTINATION_SERVICE_RESOURCE in con.source &&
+        con.source[SPAN_DESTINATION_SERVICE_RESOURCE] ===
+          node[SPAN_DESTINATION_SERVICE_RESOURCE]
+    );
     const matchedService = discoveredServices.find(({ from }) => {
-      if ('span.destination.service.resource' in node) {
+      if (
+        !outboundConnectionExists &&
```

Comment:
What does this do exactly?

Reply:
We map a `SPAN_DESTINATION_SERVICE_RESOURCE` to a downstream service ONLY in the case when there is NO explicit outbound connection to the `SPAN_DESTINATION_SERVICE_RESOURCE` (i.e. a `queue` or so).

Reply:
Sorry for not following up here. It hurts my brain a little to figure out what happens here as we have multiple de-duplication phases, and adding messaging queues happens somewhere in between, but I've been trying to think of edge cases for a bit and cannot quickly think of any, so let's just move forward with this and use it (my trouble of understanding how it works now) as another signal that this is desperately in need of a rewrite 😄